Streamline documentation to reflect project "piper" (#875)
SAP/jenkins-library: documentation/docs/index.md

# Project "Piper" User Documentation

Continuous delivery is a method to develop software with short feedback cycles.
It is applicable to projects both for SAP Cloud Platform and SAP on-premise platforms.
SAP implements tooling for continuous delivery in project "Piper".
The goal of project "Piper" is to substantially ease setting up continuous delivery in your project using SAP technologies.

## What you get

To get you started quickly, project "Piper" offers you the following artifacts:

> * A set of ready-made Continuous Delivery pipelines for direct use in your project

Reviewer: ```suggestion
* A set of preconfigured Continuous Delivery pipelines for direct use in your project
```

Author: "ready-made" is not a nice word, but "preconfigured" does not really describe what we want to convey.. What I want to say is that they are already coded, and you as the consumer don't have to code your own pipeline. Can you suggest a better wording for this, @SarahNoack?

Reviewer 2: Pre-configured does not properly address the added value. So far use "ready-made" or "out-of-the-box". If that's something we should change, it needs to be addressed holistically.

Reviewer 3: I would got with "ready-made" as it means "von der Stange" and that is from my understanding what we would like to express.

    * [General Purpose Pipeline](stages/introduction/)
    * [SAP Cloud SDK Pipeline][cloud-sdk-pipeline]

> * [A shared library][piper-library] of reusable step implementations, enabling you to customize our ready-made pipelines, or to even build your own customized pipeline

Reviewer: ```suggestion
* [A shared library][piper-library] that contains reusable step implementations, which enable you to customize our preconfigured pipelines, or to even build your own customized ones
```

> * A set of [Docker images][devops-docker-images] to setup a ci/cd environment in minutes using sophisticated life-cycle management

Reviewer: ```suggestion
* A set of [Docker images][devops-docker-images] to setup a CI/CD environment in minutes using sophisticated life-cycle management
```

> To find out which offering is right for you, we recommend to look at the ready-made pipelines first.

Reviewer: ```suggestion
To find out which offering is right for you, we recommend to look at the preconfigured pipelines first.
```

In many cases, they should satisfy your requirements, and if this is the case, you don't need to build your own pipeline.

> ### The best-practice way: Ready-made pipelines

Reviewer: ```suggestion
### The best-practice way: Preconfigured pipelines
```

**Are you building a standalone SAP Cloud Platform application?<br>**

> Then continue reading about our [general purpose pipeline](stages/introduction/) which supports various technologies and programming languages.

Reviewer: ```suggestion
Then continue reading about our [general purpose pipeline](stages/introduction/), which supports various technologies and programming languages.
```

> **Are you building an application with the SAP Cloud SDK?<br>**

Reviewer 4: and SAP cloud Application Prgammign Model

Then we can offer you a [pipeline specifically tailored to SAP Cloud SDK applications][cloud-sdk-pipeline]

### The do-it-yourself way: Build with Library

The shared library contains building blocks for your own pipeline, following our best practice Jenkins pipelines described in the Scenarios section.

The best practice pipelines are based on the general concepts of [Pipelines as Code, as introduced in Jenkins 2][jenkins-doc-pipelines].
With that you have the power of the Jenkins community at hand to optimize your pipelines.

You can run the best practice Jenkins pipelines out of the box, take them as a
starting point for project-specific adaptations or implement your own pipelines
from scratch using the shared library.

> For an example, you might want to check out our ["Build and Deploy SAPUI5 or SAP Fiori Applications on SAP Cloud Platform with Jenkins" scenario][piper-library-scenario].

Reviewer 4: Also the CAP one ...
